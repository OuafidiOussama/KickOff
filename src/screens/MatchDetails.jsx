import { View, Text } from "react-native";
import React from "react";
import { useSelector } from "react-redux";
import MDetailsCard from "../components/cards/MDetailsCard";

export default function MatchDetails() {
  const { status, match, players } = useSelector((state) => state.Matches);
  if (status === "loading") {
    return (
      <View>
        <Text>Loading ...</Text>
      </View>
    );
  } else if (status === "success") {
    return <MDetailsCard match={match} players={players} />;
  } else {
    return (
      <View>
        <Text>Failed</Text>
      </View>
    );
  }
}
