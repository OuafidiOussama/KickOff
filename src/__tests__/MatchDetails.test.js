import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import MatchDetails from "../screens/MatchDetails";

const mockStore = configureStore([]);

describe("<MatchDetails />", () => {
  it("renders loading text when status is loading", () => {
    const initialState = {
      Matches: { status: 'loading', match: null, players: [] }
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <MatchDetails />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  it("renders failed text when status is not loading, not success", () => {
    const initialState = {
      Matches: { status: 'failed', match: null, players: [] }
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <MatchDetails />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
