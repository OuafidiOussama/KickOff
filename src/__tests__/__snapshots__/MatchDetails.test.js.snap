// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchDetails /> renders failed text when status is not loading or success 1`] = `
<View>
  <Text>
    Failed
  </Text>
</View>
`;

exports[`<MatchDetails /> renders loading text when status is loading 1`] = `
<View>
  <Text>
    Loading ...
  </Text>
</View>
`;
